google/resource_compute_shared_vpc_host.go

```diff
@@ -0,0 +1,91 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceComputeSharedVpcHost() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeSharedVpcHostCreate,
+		Read:   resourceComputeSharedVpcHostRead,
+		Delete: resourceComputeSharedVpcHostDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceComputeSharedVpcHostCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	projectID, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	op, err := config.clientCompute.Projects.EnableXpnHost(projectID).Do()
```

Comment:
What is the behavior of the EnableXpnHost API call if Xpn is already enabled? We should make sure that we don't fail the create operation if xpnHost is already enabled.

Reply:
It succeeds!